src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -0,0 +1,54 @@
+import { BigNumber } from "ethers";
+import { HubPoolClient } from "../HubPoolClient";
+import { ERC20__factory, toBN } from "@across-protocol/contracts-v2";
+import { calculateUtilizationBoundaries, computePiecewiseLinearFunction } from "../../UBAFeeCalculator/UBAFeeUtility";
+import { SpokePoolClient } from "../SpokePoolClient";
+import { FlowTupleParameters } from "../../UBAFeeCalculator/UBAFeeConfig";
+import { abs, max } from "../../utils";
+
+/**
+ * Compute the realized LP fee for a given amount.
+ * @param l1TokenAddress The L1 token address to get the LP fee
+ * @param depositChainId The chainId of the deposit
+ * @param refundChainId The chainId of the refund
+ * @param amount The amount that is being deposited
+ * @param hubPoolClient A hubpool client instance to query the hubpool
+ * @param spokePoolClients A mapping of spoke chainIds to spoke pool clients
+ * @param baselineFee The baseline fee to use for this given token
+ * @param gammaCutoff The gamma cutoff to use for this given token - used in the piecewise linear function calculation
+ * @returns The realized LP fee for the given token on the given chainId at the given block number
+ */
+export async function computeRealizedLpFeeForRefresh(
+  l1TokenAddress: string,
+  depositChainId: number,
+  refundChainId: number,
+  amount: BigNumber,
+  hubPoolClient: HubPoolClient,
+  spokePoolClients: { [chainId: number]: SpokePoolClient },
+  baselineFee: BigNumber,
+  gammaCutoff: FlowTupleParameters,
+  blockNumber?: number
```

Comment:
Why make `blockNumber` optional? Isn't this always required?

Reply:
The default undefined block number polls the RPC with the `"latest"` block number

Reply:
oh ok